File: python_backend/pathsInfo/get_index_first_piece.py
from custom_except import *

def get_index_first_piece(ids):
    """take a list of ids and return the index of the first piece id"""

    for i in range(len(ids)):
        if ids[i] != '#':
            return i
    else:
        return 'no pieces'


if __name__ == "__main__":
     i = get_index_first_piece(['#', 'WP1', 'WP2', 'BB1'])
     print(i)

File: app/src/game_logic/pathsInfo/getIndexFirstPiece.js


export function getIndexFirstPiece(ids) {
    /*take a list of assignIds &&  return the index of the first piece id**/
    for (let i = 0; i < ids.length; i++) {
        if (ids[i] !== '#') {
            return i
        }
    }
    return 'no pieces'
}

// module.exports = getIndexFirstPiece;

/** for node.js
if (require.main === module) {
    var i = getIndexFirstPiece(['#', 'WP1', 'WP2', 'BB1'])
    console.log(i)
    i = getIndexFirstPiece(['#', '#', '#', 'BB1'])
    console.log(i)

}
*/

